tests/functionaltests/Tests/CoreAnimationTests/CALayerAppearanceTests.mm

```diff
@@ -104,7 +106,7 @@ - (void)viewDidLoad {
     });
 
     [condition lock];
-    ASSERT_TRUE_MSG([condition waitUntilDate:[NSDate dateWithTimeIntervalSinceNow:c_testTimeoutInSec]],
+    ASSERT_TRUE_MSG(signaled || [condition waitUntilDate:[NSDate dateWithTimeIntervalSinceNow:c_testTimeoutInSec]],
```

Comment:
It's the proper design when using NSCondition as per Apple doc ([https://developer.apple.com/reference/foundation/nscondition](https://developer.apple.com/reference/foundation/nscondition)). In this example, if the property change callback is fired before we get to the wait state, we can conclude the test at that point using the signaled predicate.

---
In reply to: [94669086](https://github.com/Microsoft/WinObjC/pull/1627#discussion_r94669086) [](ancestors = 94669086)